builder/beacon_client.go

```diff
@@ -51,14 +62,56 @@ func (b *BeaconClient) isValidator(pubkey PubkeyHex) bool {
 }
 
 func (b *BeaconClient) getProposerForNextSlot(requestedSlot uint64) (PubkeyHex, error) {
-	/* Only returns proposer if requestedSlot is currentSlot + 1, would be a race otherwise */
 	b.mu.Lock()
 	defer b.mu.Unlock()
 
-	if b.currentSlot+1 != requestedSlot {
-		return PubkeyHex(""), errors.New("slot out of sync")
+	if (requestedSlot / b.slotsInEpoch) > b.currentEpoch {
```

Comment:
Hm, since `b.currentEpoch` is set elsewhere, calling this function twice with the same `requestedSlot` (which is possible) causes `currEpochProposerMap` to be cycled twice which might be incorrect

Reply:
Call it twice in a row with the same requestedSlot and the asignment to currEpochProposerMap will happen twice as currentEpoch is not set in this function